### PR TITLE
feat: Add vpc cni metric helper security group ingress rule

### LINKED ISSUE #2394: Unable to use cni-metric-helper with default configuration
## Description

I'm unable to use the [cni-metric-helper](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/cmd/cni-metrics-helper/README.md] 

The default config for security group ingress rule doesn't allow port 61678 from cluster to node

If your request is for a new feature, please use the `Feature request` template.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Reproduction Code [Required]

Install eks cluster with default config
Install cni-metric-helper following instructions https://github.com/terraform-aws-modules/terraform-aws-eks/

In my case the cni-metric-helper was not able to communicate and printed the following logs
```
Failed to grab CNI endpoint: the server is currently unable to handle the request (get pods aws-node-ktpbn:61678)"}
Failed to grab CNI endpoint: the server is currently unable to handle the request (get pods aws-node-4tjv5:61678)"}
```

## Expected behavior

To be able to get metrics from vpc-cni ip address allocations

## Actual behavior

I can't get metrics

## Additional context

Issue to add more docs about this requirement https://github.com/aws/amazon-vpc-cni-k8s/issues/2208

I was able to fix the problem by adding a security group ingress rule to allow inbound to nodes from cluster for port tcp 61678

I think the code to update to fix is located here https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/node_groups.tf#L74-L106

### PR DESCRIPTION
**Pull request by csantanapr**
## Description

Adds VPC CNI Metric security group ingress rule with port 61678

## Motivation and Context

Resolves #2394 

## Breaking Changes
No braking change

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by csantanapr** [comment]
Moving to recommended section along with the other addons

**Comment by csantanapr** [comment]
moved to recommended section where we specify ALB and Karpenter, metric-server this will add VPC CNI. VPC CNI is not an user workload so I think we should added

File: node_groups.tf [hunk 1]
```diff
@@ -150,6 +142,15 @@ locals {
       type                          = "ingress"
       source_cluster_security_group = true
     }
+    # VPC CNI
+    ingress_cluster_vpc_cni_metrics = {
+      description                   = "Cluster API to node 61678/tcp VPC CNI metrics"
+      protocol                      = "tcp"
+      from_port                     = 61678
```
Comment (bryantbiggs) [inline node_groups.tf:149]: The "recommended" rules are those that are most commonly found in clusters - there isn't a set rule of what makes it "common" or "recommended", but based off of historical issues raised by users. The VPC CNI metrics and its associated access/port has not come up so I don't believe this falls under the recommended category. 

If a user wishes to use this addon and open up this network connectivity, they can do so by providing the rule in their configuration similar to shown here https://github.com/terraform-aws-modules/terraform-aws-eks/blob/674ab7b7f2aa5163b3d9dbdafb7cb1ec910cd9de/examples/complete/main.tf#L78-L117

**Comment by bryantbiggs** [comment]
Thank you but at this time, we will not be adding this to the recommended security group rules

**Pull request closed without merging**